plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/auth/SMSessionUtils.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * DBeaver - Universal Database Manager
+ * Copyright (C) 2010-2023 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jkiss.dbeaver.model.auth;
+
+import org.jkiss.dbeaver.DBException;
+import org.jkiss.dbeaver.model.app.DBPWorkspace;
+import org.jkiss.dbeaver.model.runtime.DBRProgressMonitor;
+import org.jkiss.dbeaver.runtime.DBWorkbench;
+
+public class SMSessionUtils {
+
+    public static SMSession getSpaceSession(DBRProgressMonitor monitor) throws DBException {
+        DBPWorkspace workspace = DBWorkbench.getPlatform().getWorkspace();
```

Comment:
We have identical code in EnterpriseWorkbenchWindowAdvisor and in QMUtils.
Maybe it is better to move it to QMUtils and make more generic?

Reply:
I extracted this method from EnterpriseWorkbenchWindowAdvisor to utils class. I don't think that it's good to add method to QMUtils because SMSession is not related to QM.